crates/uv/src/lib.rs

```diff
@@ -168,45 +168,67 @@ async fn run(mut cli: Cli) -> Result<ExitStatus> {
+                ) => match Pep723Script::read(&script).await {
+                    Ok(Some(script)) => Some(Pep723Item::Script(script)),
+                    Ok(None) => None,
+                    Err(Pep723Error::Io(err)) if err.kind() == std::io::ErrorKind::NotFound => None,
+                    Err(err) => return Err(err.into()),
+                },
                 Some(RunCommand::PythonRemote(script, _)) => {
-                    Pep723Metadata::read(&script).await?.map(Pep723Item::Remote)
+                    match Pep723Metadata::read(&script).await {
+                        Ok(Some(metadata)) => Some(Pep723Item::Remote(metadata)),
+                        Ok(None) => None,
+                        Err(Pep723Error::Io(err)) if err.kind() == std::io::ErrorKind::NotFound => {
+                            None
+                        }
+                        Err(err) => return Err(err.into()),
+                    }
                 }
                 Some(
                     RunCommand::PythonStdin(contents, _) | RunCommand::PythonGuiStdin(contents, _),
                 ) => Pep723Metadata::parse(contents)?.map(Pep723Item::Stdin),
                 _ => None,
-            }
-        } else if let ProjectCommand::Remove(uv_cli::RemoveArgs {
-            script: Some(script),
-            ..
-        }) = &**command
-        {
-            Pep723Script::read(&script).await?.map(Pep723Item::Script)
-        } else if let ProjectCommand::Lock(uv_cli::LockArgs {
-            script: Some(script),
-            ..
-        }) = &**command
-        {
-            Pep723Script::read(&script).await?.map(Pep723Item::Script)
-        } else if let ProjectCommand::Tree(uv_cli::TreeArgs {
-            script: Some(script),
-            ..
-        }) = &**command
-        {
-            Pep723Script::read(&script).await?.map(Pep723Item::Script)
-        } else if let ProjectCommand::Export(uv_cli::ExportArgs {
-            script: Some(script),
-            ..
-        }) = &**command
-        {
-            Pep723Script::read(&script).await?.map(Pep723Item::Script)
-        } else {
-            None
+            },
+            ProjectCommand::Remove(uv_cli::RemoveArgs {
+                script: Some(script),
+                ..
+            })
+            | ProjectCommand::Lock(uv_cli::LockArgs {
+                script: Some(script),
+                ..
+            })
+            | ProjectCommand::Tree(uv_cli::TreeArgs {
+                script: Some(script),
+                ..
+            })
+            | ProjectCommand::Export(uv_cli::ExportArgs {
+                script: Some(script),
+                ..
+            }) => match Pep723Script::read(&script).await {
+                Ok(Some(script)) => Some(Pep723Item::Script(script)),
+                Ok(None) => {
+                    // TODO(charlie): `uv lock --script` should initialize the tag, if it doesn't
+                    // exist.
```

Comment:
@zanieb -- Just calling out that I deferred this. It looks like a lot more work.